std/algorithm/iteration.d

```diff
@@ -2948,6 +2955,20 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR))
     assert(!equal(js2, js));
 }
 
+// Bugzilla issue 19213
+@system unittest
+{
+    static struct Rec
+    {
+        int foo;
+    }
+    static assert(__traits(compiles, {
+        auto results = [[1,2], [3,4]]
+            .map!(q => q.map!Rec.chunkBy!"a.foo")
+            .joiner;
+    }));
```

Comment:
No need to use `__traits(compiles` here. Use the range directly and even `assert` that its still iterated correctly.
Also, it would be good to add a test for iterating this range backwards and checking the result.